File: app/src/main/java/com/example/hellokt/bottomBar/BottomBarController.java
package com.example.hellokt.bottomBar;

import androidx.annotation.IdRes;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentActivity;

public interface BottomBarController {
    BottomBarLayout setSourceFrameLayout(FragmentActivity activity, @IdRes int frameLayoutId);
    BottomBarLayout addItemView(BottomBarItemView view, Fragment fragment);

    boolean removeItemView(int position);

    boolean removeItemView(BottomBarItemView view);

    void setSelected(int position);

    void initialise();
}

File: app/src/main/java/com/example/hellokt/bottomBar/BottomBarController.kt
package com.example.hellokt.bottomBar

import androidx.annotation.IdRes
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentActivity

interface BottomBarController {

    fun setSourceFrameLayout(activity: FragmentActivity?, @IdRes frameLayoutId: Int): BottomBarLayout?

    fun addItemView(view: BottomBarItemView?, fragment: Fragment?): BottomBarLayout?

    fun removeItemView(position: Int): Boolean

    fun removeItemView(view: BottomBarItemView?): Boolean

    fun setSelected(position: Int)

    fun initialise()
}